File: app/src/main/java/com/example/final_pocproject/activities/PostActivity.java
package com.example.final_pocproject.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.final_pocproject.adapters.CommentListViewAdapter;
import com.example.final_pocproject.api.ApiInterface;
import com.example.final_pocproject.interfaces.PostInterface;
import com.example.final_pocproject.model.Post;
import com.example.final_pocproject.adapters.PostListViewAdapter;
import com.example.final_pocproject.R;
import com.example.final_pocproject.api.RetrofitClient;
import com.example.final_pocproject.utility.PostUtility;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class PostActivity extends AppCompatActivity implements PostInterface {

    ApiInterface apiInterface;
    PostUtility postUtility;
    private List<Post> postList;
    private ListView postListView;
    PostListViewAdapter adapter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_post);

        Intent intent = getIntent();
        int userId = intent.getIntExtra("id", 0);

        postListView = findViewById(R.id.listView);

        postUtility = new PostUtility(this);
        postUtility.getPosts(userId);


        postListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                TextView textViewId = view.findViewById(R.id.post_id);
                int postId = Integer.parseInt(String.valueOf(textViewId.getText()));
                Intent intent = new Intent(PostActivity.this, CommentActivity.class);
                intent.putExtra("postId", postId);
                startActivity(intent);
            }
        });
    }

    @Override
    public void handleSuccessResponse(List<Post> posts) {

        if (posts == null) {
            Toast.makeText(PostActivity.this, "Posts could not be loaded", Toast.LENGTH_SHORT).show();
        }
        adapter = new PostListViewAdapter(getApplicationContext(), posts);
        postListView.setAdapter(adapter);
    }

    @Override
    public void handleFailure(Throwable t) {
        Toast.makeText(PostActivity.this, t.getMessage(), Toast.LENGTH_SHORT).show();

    }
}

File: app/src/main/java/com/example/final_poc_project/activities/PostActivity.kt
package com.example.final_poc_project.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.final_poc_project.*
import com.example.final_poc_project.adapters.PostAdapter
import com.example.final_poc_project.clickListener.PostCellClickListener
import com.example.final_poc_project.interfaces.PostInterface
import com.example.final_poc_project.model.Post
import com.example.final_poc_project.api.PostApi


class PostActivity : AppCompatActivity(), PostInterface, PostCellClickListener {

    lateinit var recyclerView: RecyclerView
    lateinit var sessionManager: SessionManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_post)

        sessionManager = SessionManager(applicationContext)

        sessionManager.checkLogin()
        sessionManager.changeScreenStatus(1)

        val user: HashMap<String,String> = sessionManager.getUserDetails()
        val userId = user[SessionManager.KEY_USERID]!!

        val postUtilityKotlin = PostApi(this)
        postUtilityKotlin.getPost(Integer.parseInt(userId))

    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        val inflater: MenuInflater = menuInflater
        inflater.inflate(R.menu.main_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        sessionManager.logoutUser()
        return super.onOptionsItemSelected(item)
    }

    override fun handleSuccessResponse(posts: List<Post>) {
        recyclerView = findViewById(R.id.recyclerView)
        recyclerView.layoutManager = LinearLayoutManager(this@PostActivity)
        recyclerView.adapter = PostAdapter(posts, this)
    }

    override fun handleFailure(t: Throwable) {
        Toast.makeText(this@PostActivity, t.message, Toast.LENGTH_SHORT).show()
    }

    override fun onCellClickListener(id: Any) {
        sessionManager.createPostSession(id as Int)
        val intent = Intent(this@PostActivity, CommentActivity::class.java)
       //intent.putExtra("POST_ID", id as Int)
        startActivity(intent)
    }

}
